project/SparkBuild.scala

```diff
@@ -748,6 +739,13 @@ object Java8TestSettings {
 object TestSettings {
   import BuildCommons._
 
+  private val scalaBinaryVersion =
+    if (System.getProperty("scala-2.10") == "true") {
```

Comment:
Is it possible to read this from the POM rather than relying on a system property to be set? I'm worried that the approach used here will not work if we have to support three Scala versions (2.10, 2.11, and 2.12).